mesheryctl/internal/cli/root/system/check.go

```diff
@@ -42,7 +44,50 @@ var checkCmd = &cobra.Command{
 func RunPreflightHealthChecks(isPreRunExecution bool, subcommand string) error {
 	mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
 	if err != nil {
-		return errors.Wrap(err, "error processing config")
+		userResponse := false
+
+		//skip asking confirmation if -y flag used or host in meshconfig is already localhost
+		if utils.SilentFlag {
+			userResponse = true
+		} else {
+			// ask user for confirmation
+			userResponse = utils.AskForConfirmation("Do you want to create a new config file")
+		}
+
+		if !userResponse {
+			return err
+		}
+
+		cfgFile := utils.CfgFile
+		log.Println("Backing up " + cfgFile + " to original_config.yaml")
+		err := os.Rename(cfgFile, "original_config.yaml")
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		_, err = os.Create(cfgFile)
```

Comment:
Should we be using https://github.com/layer5io/meshery/blob/3538340cf267bc92c60eed749c857d5d28c5f89e/mesheryctl/pkg/utils/helpers.go#L588 instead?

Reply:
@Lee I have a doubt.  
 I think `CreateConfigFile` creates a file in `~/.meshery`, whereas I think it would be better to create a file in the user's original directory, with the same name, since there might already be a `meshconfig` in `~/.meshery`.  
What do you think?

Reply:
Good consideration... this is where the .bak comes in.

Reply:
Alright @leecalcote!  
So just to confirm before the commit, if a user has a buggy meshconfig:  
1. Create a `.bak` copy in their current directory.    
2. Create a new config(use the existing function) and populate it with defaults, but this time in `~/.meshery`, overwriting their existing `meshconfig,` if any, in `~/.meshery`.  

Is that the final plan then?

Reply:
@leecalcote I've used `.bak.yaml` since viper recognises only `YAML` in Meshery.